cmd/snap-confine/snap-confine.c

```diff
@@ -527,29 +602,6 @@ static void enter_non_classic_execution_environment(sc_invocation * inv,
 						    gid_t real_gid,
 						    gid_t saved_gid)
 {
-	/* snap-confine uses privately-shared /run/snapd/ns to store bind-mounted
-	 * mount namespaces of each snap. In the case that snap-confine is invoked
-	 * from the mount namespace it typically constructs, the said directory
-	 * does not contain mount entries for preserved namespaces as those are
-	 * only visible in the main, outer namespace.
-	 *
-	 * In order to operate in such an environment snap-confine must first
-	 * re-associate its own process with another namespace in which the
-	 * /run/snapd/ns directory is visible. The most obvious candidate is pid
-	 * one, which definitely doesn't run in a snap-specific namespace, has a
-	 * predictable PID and is long lived.
-	 */
-	sc_reassociate_with_pid1_mount_ns();
-	// Do global initialization:
-	int global_lock_fd = sc_lock_global();
-	// ensure that "/" or "/snap" is mounted with the
-	// "shared" option, see LP:#1668659
-	debug("ensuring that snap mount directory is shared");
-	sc_ensure_shared_snap_mount();
-	debug("unsharing snap namespace directory");
-	sc_initialize_mount_ns();
-	sc_unlock(global_lock_fd);
-
```

Comment:
Like the above, and not a blocker, but if adding things, maybe add a comment that main.c will reassociate for /run/snapd/ns before calling this? The comment doesn't have to be long, just a quick pointer.